source/yats/include/yats/TaskConfigurator.h

```diff
@@ -82,7 +82,7 @@ class TaskConfigurator : public AbstractTaskConfigurator
 	std::enable_if_t<Index < Helper::ParameterCount> parseInputParameter()
 	{
 		using currentInput = std::tuple_element_t<Index, typename Helper::WrappedInput>;
-		m_inputs.insert(std::make_pair(currentInput::ID, InputConnector(this)));
+		m_inputs.insert(std::make_pair(currentInput::ID, InputConnector<currentInput>(this)));
```

Comment:
use emplace, also consider object slicing. This is now a dangerous operation.

Reply:
I changed insert to emplace. Are we on the safe side now?

Reply:
The object slicing is still a problem here, see: http://www.geeksforgeeks.org/object-slicing-in-c/

Also I can not see your changes.